api-gateway/src/main/kotlin/com/saveourtool/save/gateway/service/BackendService.kt

```diff
@@ -64,23 +58,33 @@ class BackendService(
         }
         .toEntity<String>()
         .map {
-            objectMapper.readValue(it.body, UserDetails::class.java)
+            springUserDetailsReader.readValue(it.body)
         }
 
     /**
      * Saves a new [User] in DB
      *
-     * @param user
+     * @param source
+     * @param nameInSource
+     * @param roles
      * @return empty [Mono]
      */
-    fun createNewIfRequired(user: User): Mono<Void> = webClient.post()
-        .uri("/internal/users/new")
-        .contentType(MediaType.APPLICATION_JSON)
-        .bodyValue(user)
-        .retrieve()
-        .onStatus({ it.is4xxClientError }) {
-            Mono.error(ResponseStatusException(it.statusCode()))
-        }
-        .toBodilessEntity()
-        .then()
+    @Suppress("UnusedParameter")
+    fun createNewIfRequired(source: String, nameInSource: String, roles: List<String>): Mono<Void> {
+        // https://github.com/saveourtool/save-cloud/issues/583
+        // fixme: this sets a default role for a new user with minimal scope, however this way we discard existing role
+        // from authentication provider. In the future we may want to use this information and have a mapping of existing
+        // roles to save-cloud roles.
+        val overriddenRoles = listOf(Role.VIEWER.asSpringSecurityRole())
+        return webClient.post()
+            .uri("/internal/users/new-if-required/$source/$nameInSource")
```

Comment:
I would actually keep "/new" - it is more convenient :)
But may be other guys will suggest

Reply:
will rename back to avoid change in endpoint name (logic wasn't changed)